src/extractor/guidance/turn_handler.cpp

```diff
@@ -166,15 +166,22 @@ Intersection TurnHandler::handleThreeWayTurn(const EdgeID via_edge, Intersection
     }
     else
     {
+        const auto direction_at_one = getTurnDirection(intersection[1].turn.angle);
+        const auto direction_at_two = getTurnDirection(intersection[2].turn.angle);
         if (obvious_index == 1)
         {
             intersection[1].turn.instruction = getInstructionForObvious(
                 3, via_edge, isThroughStreet(1, intersection), intersection[1]);
         }
         else
         {
-            intersection[1].turn.instruction = {findBasicTurnType(via_edge, intersection[1]),
-                                                getTurnDirection(intersection[1].turn.angle)};
+            if (obvious_index != 0 && direction_at_one == direction_at_two &&
```

Comment:
I had a little trouble following the cases here. It is not obvious what value obvious_index is (pun not intended). The readability might be improved by first dividing control flow on that value, instead of doing it in a nested manner.

```
if (obvious index == 0)
else if (obvious_index == 1)
else {
BOOST_ASSERT (obvious_index == 2);
}
```